typescript/infra/src/config/nautilus_token_config.ts

```diff
@@ -0,0 +1,66 @@
+import { ChainMap } from '@hyperlane-xyz/sdk';
+
+interface NativeTokenConfig {
+  chainId: number;
+  symbol: string;
+  name: string;
+  type: 'native';
+  decimals: number;
+  hypNativeAddress: string;
+  protocolType?: 'ethereum' | 'solana';
+}
+
+interface CollateralTokenConfig {
+  // Typescript does not allow literal value value 'collateral' even if it matches the enum's value
+  type: 'collateral';
+  address: string;
+  chainId: number;
+  decimals: number;
+  symbol: string;
+  name: string;
+  hypCollateralAddress: string;
+  isSpl2022?: boolean;
+  protocolType?: 'ethereum' | 'solana';
+}
+
+export type WarpTokenConfig = ChainMap<
+  CollateralTokenConfig | NativeTokenConfig
+>;
```

Comment:
I'm okay with copying these config shapes from the Warp UI but please leave a TODO to migrate the warp UI configs into the SDK and de-dupe them

Reply:
Left a TODO